src/overlays/actors/ovl_En_Syateki_Niw/z_en_syateki_niw.c

```diff
@@ -21,37 +42,710 @@ const ActorInit En_Syateki_Niw_InitVars = {
+                    if (this->unk_2E8.x < -150.0f) {
+                        this->unk_2E8.x = -150.0f;
+                    }
+                    if (this->unk_2E8.x > 150.0f) {
+                        this->unk_2E8.x = 150.0f;
+                    }
+                    if (this->unk_2E8.z < -60.0f) {
+                        this->unk_2E8.z = -60.0f;
+                    }
+                    if (this->unk_2E8.z > -40.0f) {
+                        this->unk_2E8.z = -40.0f;
+                    }
+                    break;
+                case 1:
+                    sp50 = Math_Rand_CenteredFloat(50.0f);
+                    if (sp50 < 0.0f) {
+                        sp50 -= 50.0f;
+                    } else {
+                        sp50 += 50.0f;
+                    }
+                    sp4C = Math_Rand_CenteredFloat(30.0f);
+                    if (sp4C < 0.0f) {
+                        sp4C -= 30.0f;
+                    } else {
+                        sp4C += 30.0f;
+                    }
+                    this->unk_2E8.x = this->unk_2DC.x + sp50;
+                    this->unk_2E8.z = this->unk_2DC.z + sp4C;
+                    break;
+            }
+        } else {
+            this->unk_25C = 4;
+            if (this->actor.bgCheckFlags & 1) {
+                this->actor.velocity.y = 2.5f;
+                if ((Math_Rand_ZeroFloat(10.0f) < 1.0f) && (this->unk_29E == 0)) {
+                    this->unk_25C = 0xC;
+                    this->actor.velocity.y = 10.0f;
+                }
+            }
+        }
+    }
+    if (this->unk_25C != 0) {
+        sp4A = 1;
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.x, this->unk_2E8.x, 1.0f, this->unk_2C8.y);
+        Math_SmoothScaleMaxF(&this->actor.posRot.pos.z, this->unk_2E8.z, 1.0f, this->unk_2C8.y);
+        Math_SmoothScaleMaxF(&this->unk_2C8.y, 3.0f, 1.0f, 0.3f);
+        tmpf1 = this->unk_2E8.x - this->actor.posRot.pos.x;
+        tmpf2 = this->unk_2E8.z - this->actor.posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Niw/func_80B12460.s")
+        if (fabsf(tmpf1) < 10.0f) {
+            tmpf1 = (f64)0.0f;
+        }
+        if (fabsf(tmpf2) < 10.0f) {
+            tmpf2 = 0.0f;
+        }
+        if ((tmpf1 == 0.0f) && (tmpf2 == 0.0f)) {
+            this->unk_25C = 0;
+            this->unk_294 = 7;
+        }
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, Math_atan2f(tmpf1, tmpf2) * 10430.378f, 3, this->unk_2C8.z,
+                                0);
+        Math_SmoothScaleMaxF(&this->unk_2C8.z, 10000.0f, 1.0f, 1000.0f);
+    }
+    if (this->unk_260 == 0) {
+        func_80B11A94(this, globalCtx, sp4A);
+        return;
+    }
+    if ((globalCtx->gameplayFrames % 4) == 0) {
+        sp7C.y = Math_Rand_CenteredFloat(5.0f);
+        sp70.y = 0.2f;
+        sp5C = this->actor.posRot.pos;
+        func_8002836C(globalCtx, &sp5C, &sp7C, &sp70, &sp6C, &sp68, 0x258, 0x28, 0x1E);
```

Comment:
id name the vectors and colors based of the prototype. dustPos dustVelocity dustAccel etc, and make the hex numbers decimal